### ISSUE TITLE
Timestamps are not imported for schemas with SQL defined default values

### ISSUE THREAD

**Issue by Andreis13** [zdennis/activerecord-import#426]
On some of our projects, schema files are defined in SQL format instead of vanilla `schema.rb`, and it is customary for such setups to have default values on timestamp columns that are defined in the schema itself:

```sql
`created_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
`updated_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
```

In this case, `read_attribute_before_type_cast` will return `"CURRENT_TIMESTAMP"` for an unsaved record:
```ruby
Record.new.read_attribute_before_type_cast(:created_at) #=> "CURRENT_TIMESTAMP"
```

As a result, when the code reaches [this line](https://github.com/zdennis/activerecord-import/blob/master/lib/activerecord-import/import.rb#L788), the `array_of_attributes` is not updated with the `created_at` timestamp, because `arr[index].nil?` returns `false` (`arr[index] == "CURRENT_TIMESTAMP"` at this point). Therefore, this will import `NULL` into this column and fail because of the `NOT NULL` constraint.

**Comment by jkowens** [comment]
Ah interesting. So after looking into this issue, it seems we'd probably have to use `read_attribute` instead of `read_attribute_before_type_cast` for any column that has a default value defined? It looks like PostgreSQL allows a function or expression to be set as a default value even for columns other than dates/datetimes. Does that seem like the only safe option?

**Comment by Andreis13** [comment]
I don't have enough context on the behaviour of various RDBMs, in order to assess the 'safety' of this solution.. But I was wondering what is the performance difference between using `read_attribute` vs `read_attribute_before_type_cast`? Maybe it makes sense to use the former for all columns regardless of type? (might greatly simplify the existing logic)

**Comment by jkowens** [comment]
From what I've seen `read_attribute_before_type_cast` is significantly faster. Here was one benchmark:

https://gist.github.com/aquajach/f875e13773ace44d6f85

I've run benchmarks where the difference is even greater.

**Comment by Andreis13** [comment]
Well yeah, the method itself is surely faster, but the question is what's the difference when used in the context of `AR::Import`? I doubt it's the bottleneck of the whole process. Say, if inserting 1000 records takes 0.5 seconds, while reading 1000 times ~25 attributes with typecasting is only 0.005 seconds faster than without typecasting, we could say that this difference is negligible.

Moreover, since the slowest typecasting operations are the ones of serialization and date parsing, and we have to typecast these types anyway, I don't see the point skipping typecasting for `int -> int` or `string -> string`.

**Comment by jkowens** [comment]
That's true, you make an excellent point.

**Comment by jkowens** [comment]
I just committed an update to use `read_attribute` for columns with default values. Eventually we could very well end up just always using `read_attribute`.

**Comment by Andreis13** [comment]
Awesome, I'll give it a try and check how it runs on our setup!

**Comment by jkowens** [comment]
Is everything all good? If so, I'll cut a new release.

**Comment by Andreis13** [comment]
Didn't reach production yet, but tests look good, thanks a lot!

**Comment by jkowens** [comment]
Awesome, thanks for the feedback 👍

**Comment by jkowens** [comment]
This fix has been released in v0.19.0.
